cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_collaborators.spec.js

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { MiscUtils } from '@opensearch-dashboards-test/opensearch-dashboards-test-library';
+const miscUtils = new MiscUtils(cy);
+const workspaceName = 'test_workspace_collaborators';
+let workspaceId;
+
+if (Cypress.env('WORKSPACE_ENABLED')) {
+  describe('Workspace collaborators', () => {
+    before(() => {
+      cy.deleteWorkspaceByName(workspaceName);
+      cy.createWorkspace({
+        name: workspaceName,
+        features: ['use-case-observability'],
+        settings: {
+          permissions: {
+            library_write: { users: ['%me%'] },
+            write: { users: ['%me%'] },
+          },
+        },
+      }).then((value) => (workspaceId = value));
+    });
+
+    beforeEach(() => {
+      // Visit workspace collaborators page
+      miscUtils.visitPage(`w/${workspaceId}/app/workspace_collaborators`);
+    });
+
+    after(() => {
+      cy.deleteWorkspaceById(workspaceId);
+    });
+
+    it('should successfully load the page', () => {
+      if (Cypress.env('SAVED_OBJECTS_PERMISSION_ENABLED')) {
+        cy.contains('Collaborators', { timeout: 60000 }).should('be.visible');
+      }
+    });
+
+    if (
+      Cypress.env('SAVED_OBJECTS_PERMISSION_ENABLED') &&
+      Cypress.env('SECURITY_ENABLED')
+    ) {
+      describe('Update collaborators', () => {
+        it('should add user and group collaborators successfully', () => {
+          //Add user
+          cy.getElementByTestId('add-collaborator-popover').click();
+          cy.get('button[id="user"]').click();
+          cy.contains('Add Users').should('be.visible');
+          cy.getElementByTestId('workspaceCollaboratorIdInput-0').type(
+            'read_user'
+          );
+          cy.get('button[type="submit"]')
+            .contains('span', 'Add collaborators')
+            .click();
+
+          //Add group
+          cy.getElementByTestId('add-collaborator-popover').click();
+          cy.get('button[id="group"]').click();
+          cy.contains('Add Groups').should('be.visible');
+          cy.getElementByTestId('workspaceCollaboratorIdInput-0').type(
+            'admin_group'
+          );
+          cy.get('span[title="Admin"]').click();
+          cy.get('button[type="submit"]')
+            .contains('span', 'Add collaborators')
+            .click();
+
+          cy.wait(2000); // Intentional Wait
```

Comment:
What about we doing assertion here instead of waiting for 2s, it seems unreliable.

Reply:
This is mainly used for waiting page load after navigation or wait stored data update after operation.